### PR TITLE
Resizing issues

### PR DESCRIPTION
**Pull request by johnkerl**
Findings from debug on https://github.com/TileDB-Inc/TileDB/pull/3620

### CODE REVIEW

File: tiledb/core.cc [hunk 1]
```diff
@@ -801,9 +801,6 @@ class PyQuery {
     tiledb_datatype_t type;
     uint32_t cell_val_num;
     std::tie(type, cell_val_num) = buffer_type(name);
-    uint64_t cell_nbytes = tiledb_datatype_size(type);
```
Comment (johnkerl) [inline tiledb/core.cc:804]: Unused, revealed by a compiler warning -- not related to this PR

File: tiledb/core.cc [hunk 2]
```diff
@@ -988,6 +985,17 @@ class PyQuery {
       buf.data_vals_read += data_vals_num;
       buf.offsets_read += offset_elem_num;
       buf.validity_vals_read += validity_elem_num;
+
+      if ((ssize_t)(buf.data_vals_read * buf.elem_nbytes) >
+          (ssize_t)buf.data.size()) {
+        throw TileDBError("data buffer out of bounds: " + name);
```
Comment (ihnorton) [inline tiledb/core.cc:991]: Would you mind to add `After read query, ` before `data buffer` here and below?

**Review by ihnorton — APPROVED** [review]
Wording update request for the error message, then LGTM. Thanks

File: tiledb/core.cc (hunk 1 above)
Comment (ihnorton) [reply tiledb/core.cc:804]: Thanks

File: tiledb/core.cc [hunk 3]
```diff
@@ -988,6 +985,17 @@ class PyQuery {
       buf.data_vals_read += data_vals_num;
       buf.offsets_read += offset_elem_num;
       buf.validity_vals_read += validity_elem_num;
+
+      if ((ssize_t)(buf.data_vals_read * buf.elem_nbytes) >
```
Comment (nguyenv) [inline tiledb/core.cc:989]: Use `Py_ssize_t` instead of `ssize_t`. This should fix the build errors on Windows, Python 3.10.

**Review by nguyenv — CHANGES_REQUESTED** [review]

**Pull request merged**